src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageDirectiveFeature.cs

```diff
@@ -9,20 +9,41 @@ namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
 {
     public static class PageDirectiveFeature
     {
-        public static bool TryGetRouteTemplate(RazorProjectItem projectItem, out string template)
+        public static bool TryGetPageDirective(RazorProjectItem projectItem, out string template)
         {
+            if (projectItem == null)
+            {
+                throw new ArgumentNullException(nameof(projectItem));
+            }
+
             const string PageDirective = "@page";
 
-            string content;
-            using (var streamReader = new StreamReader(projectItem.Read()))
+            var stream = projectItem.Read();
+
+            string content = null;
+            using (var streamReader = new StreamReader(stream))
             {
-                content = streamReader.ReadToEnd();
+                do
+                {
+                    content = streamReader.ReadLine();
+                } while (content != null && string.IsNullOrWhiteSpace(content));
+                content = content != null ? content.Trim() : content;
             }
 
-            if (content.StartsWith(PageDirective, StringComparison.Ordinal))
+            if (content != null && content.StartsWith(PageDirective, StringComparison.Ordinal))
             {
-                var newLineIndex = content.IndexOf(Environment.NewLine, PageDirective.Length);
-                template = content.Substring(PageDirective.Length, newLineIndex - PageDirective.Length).Trim();
+                template = content.Substring(PageDirective.Length, content.Length - PageDirective.Length).Trim();
+
+                if (template.StartsWith("\"") && template.EndsWith("\""))
+                {
+                    template = template.Substring(1, template.Length - 2);
+                }
+                // If it's not in quotes it's not our template
+                else
+                {
+                    template = string.Empty;
```

Comment:
This is an interesting case. If a user writes: `@page foo` in `bar.cshtml` we'll map the Razor page to `/bar`. However, the user will then try to access their Razor page at `/bar/foo` with no luck (it'll 404). Ideally we could raise `@page` errors on startup so they could be fixed without a user wondering why they can't hit their page.

Not much we can do in this PR so mostly just throwing this out to thinking about it. What's everyone's thoughts?